### DISCUSSION TITLE
Address normalization / internationalization

### DISCUSSION THREAD

**Discussion by rrrnld** [saleor/saleor#16628] [Q&A]
Hi there,

Does [disabling address normalization](https://docs.saleor.io/api-reference/checkout/inputs/checkout-address-validation-rules#checkoutaddressvalidationrulesenablefieldsnormalizationboolean--) skip internationalization?

For example: If I enter an address in Kyoto, Japan, I want the region and city to be spelled exactly as entered (in this case: using the latin alphabet). I ran some tests directly against the GraphQL API and the region seems to be normalized to Kanji, regardless of whether I use 

```
validationRules: {
  enableFieldsNormalization: false
}
```

when setting the checkout address or not. Is this a misunderstanding on my side? 

this is the full mutation i'm using:

```graphql
mutation checkoutAddressUpdate(
  $checkoutToken: UUID!
  $addShippingAddress: Boolean!
  $shippingAddress: AddressInput!
  $billingAddress: AddressInput!
  $validationRules: CheckoutAddressValidationRules
) {
  checkoutShippingAddressUpdate(
    shippingAddress: $shippingAddress
    validationRules: $validationRules
    token: $checkoutToken
  ) @include(if: $addShippingAddress) {
    errors {
      field
      message
      code
    }
    checkout {
      shippingMethods {
        id
      }
    }
  }
  checkoutBillingAddressUpdate(
    billingAddress: $billingAddress
    validationRules: $validationRules
    token: $checkoutToken
  ) {
    errors {
      ...CheckoutError
    }
  }
}
```

**Comment by IKarbowiak** [comment]
Hi @rrrnld,
Could you provide an example `AddressInput` that generates an invalid response?

**Reply by rrrnld** [reply]
Hi @IKarbowiak, there is no invalid response. It always responds successfully. But if I set my `AddressInput`  like this:

``` graphql
{
  firstName: "Foo"
  lastName: "Bar"
  streetAddress1: "1-1 Nakano-machi"
  streetAddress2: "Kogakuin University"
  country: "JA"
  postalCode: "192-0015"
  countryArea: "Tokyo"
}
```

The `countryArea` will be stored in the database as `東京都` despite setting `enableFieldsNormalization: false`. Is this intentional?Other fields are not normalized (`streetAddress1`), but I don't know if that's because of the `enableFieldsNormalization` setting or because the i18n library does not attempt to do so regardless.

**Reply by IKarbowiak** [reply]
Which version are you using? I checked this on the `main` branch and I got proper results. When `enableFieldsNormalization` is set to `False` the `countryArea` is not changed, when it's `True` it's changed to `東京都`

<img width="992" alt="image" src="https://github.com/user-attachments/assets/5d6a91e9-a638-404a-8ccd-8019cfb313db">
<img width="1006" alt="image" src="https://github.com/user-attachments/assets/8dfac62b-ff38-4339-b345-18c22c0e0f0a">
